(DOCSP-26259): Kotlin Create & Manage User API Keys (#2492)
mongodb/docs-realm: source/sdk/kotlin/app-services/manage-user-api-keys.txt

.. _kotlin-manage-user-api-keys:

=================================
Manage User API Keys - Kotlin SDK
=================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

> You can use the Kotlin SDK to create and manage :ref:`user API keys

Reviewer: ```suggestion
You can use the Kotlin SDK to create and manage :ref:`User API keys
```

Author: Fixed

<api-key-authentication-user-api-key>` that allow devices or services to 
communicate with Realm on behalf of an existing user without requiring the user
to re-authenticate.

User API keys are associated with a user object created by another
non-anonymous authentication provider. Each user can associate up to 20
user keys with their account. 

You can manage keys through the user's 
`apiKeyAuth <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.auth/-api-key-auth/index.html>`__

> instance.

Reviewer: Needed fix: I don't think it makes sense to say "You can manage keys through the user's instance" because "instance" implies that it is an instance of a class, but "apiKeyAuth" is an interface.

Author: Reworded. Thanks for catching!

To find a specific user API key for the logged-in user, pass the key's ``id`` 
to 
`ApiKeyAuth.fetch() <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.auth/-api-key-auth/fetch.html>`__.

.. literalinclude:: /examples/generated/kotlin/UserAPIKeysTest.snippet.fetch-api-key.kt
    :language: kotlin

.. _kotlin-enable-disable-user-key:

Enable or Disable a User API Key
--------------------------------

You can enable or disable a key by calling 
`ApiKeyAuth.enable() <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.auth/-api-key-auth/enable.html>`__ 
or 
`ApiKeyAuth.disable() <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.auth/-api-key-auth/disable.html>`__
with the key's ``id``.
Disabled keys cannot be used to log in on behalf of the user.

.. literalinclude:: /examples/generated/kotlin/UserAPIKeysTest.snippet.enable-api-key.kt
    :language: kotlin

.. _kotlin-delete-user-key:

Delete a User API Key
---------------------

To permanently remove a specific user API key, pass the
key's ``id`` to   
`ApiKeyAuth.delete() <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.auth/-api-key-auth/delete.html>`__.

> Deleted keys cannot be recovered or used to log in on behalf of the user.

Reviewer: ```suggestion
You cannot recover deleted keys or use them to log in on behalf of the user.
```
Suggestion: to avoid passive voice

Author: Reqorded here and in the 'disable key' section.
